src/components/PopupWithForm.js

```diff
@@ -2,32 +2,44 @@ import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
   constructor(popupSelector, handleFormSubmit) {
-    super({popupSelector});
+    super({ popupSelector });
     this._handleFormSubmit = handleFormSubmit;
-    this._form = this._popupElement.querySelector('.modal__form');
+    this._form = this._popupElement.querySelector(".modal__form");
   }
 
   _getInputValues() {
     const inputValues = {};
     this._inputList = this._form.querySelectorAll(".modal__input");
-    this._inputList.forEach(input => {
-      inputValues[input.name] = input.value;  //(???)
+    this._inputList.forEach((input) => {
+      inputValues[input.name] = input.value; //(???)
     });
     return inputValues;
   }
 
-  setEventListeners(){
+  setEventListeners() {
     super.setEventListeners();
-    this._form.addEventListener('submit', (evt) => {
+    this._form.addEventListener("submit", (evt) => {
       evt.preventDefault();
       this._handleFormSubmit(this._getInputValues());
       this.close();
     });
   }
 
-close() {
+  close() {
     super.close();
   }
 
-}
+  setSubmit(handleSubmit) {
+    this._handleFormSubmit = handleSubmit;
+  }
 
+  renderLoading(isSaving) {
+    this._modalButton = this._popupElement.querySelector(".modal__button");
+    if(isSaving) {
+      this._modalButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```